### PR TITLE
Convert Repo file fragment to Compose

### PR DESCRIPTION
**Pull request by VanyaHuamanBottleRocket**

### CODE REVIEW

**Review by BottleRocket-Colin — APPROVED** [review]

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt [hunk 1]
```diff
@@ -1,13 +1,112 @@
...
+        )
+        RawFileData(byteArray = state.content.value)
+    }
+}
+
+@Composable
+fun RawFileData(byteArray: ByteArray?) {
+    byteArray?.let {
+        val imageBitmap = byteArray.convertToImageBitmap()
+        if (imageBitmap != null) {
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt:71]: @VanyaHuamanBottleRocket Nice! Thanks for adding this support :)

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowserMockData.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    "*.mp4\n" +
+    "*.tiff\n" +
+    "*.avi\n" +
+    "*.flv\n" +
+    "*.mov\n" +
+    "*.wmv\n"
+
+
+internal val fileBrowserMockData = FileBrowserScreenState(
+    path = mutableStateOf(".gitingore"),
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowserMockData.kt:57]: @VanyaHuamanBottleRocket nit: Change to `.gitignore`

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowserMockData.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
+package com.bottlerocketstudios.compose.repository
+
+import androidx.compose.runtime.mutableStateOf
+
+internal val textString: String = "# These are some examples of commonly ignored file patterns.\n" +
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowserMockData.kt:5]: @VanyaHuamanBottleRocket Make this `private` if only used in this file OR refactor the name to be more specific and mentioning that it is mock data if it is used in other files

File: compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt [hunk 4]
```diff
@@ -1,13 +1,112 @@
...
+                top = Dimens.grid_1
+            )
+            .fillMaxWidth()
+            .verticalScroll(scroll)
+    )
+}
+
+@Preview(showSystemUi = true)
+@Composable
+fun FileBrowserScreenPreview() {
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/repository/FileBrowser.kt:106]: @VanyaHuamanBottleRocket Would it be possible to reference image data in another preview so both renditions could be displayed?

File: compose/src/main/java/com/bottlerocketstudios/compose/util/BinaryExtUtil.kt [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+import android.graphics.BitmapFactory
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.graphics.asImageBitmap
+
+fun ByteArray.convertToImageBitmap(): ImageBitmap? {
+    return try {
+        BitmapFactory.decodeByteArray(this, 0, this.count()).asImageBitmap()
+    } catch (e: Exception) {
+        null
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/util/BinaryExtUtil.kt:11]: @VanyaHuamanBottleRocket Let's log the exception with Timber to aid in future debugging/error understanding

**Review by jaredstockton — COMMENTED** [review]
@VanyaHuamanBottleRocket Nice work! I did leave some post-merge comments. It would be awesome if you (or somebody else) could tackle them and create a new PR :)

cc: @BottleRocket-Colin

**Pull request merged**